src/HasteModuleLoader/HasteModuleLoader.js

```diff
@@ -485,11 +487,28 @@ Loader.prototype._nodeModuleNameToPath = function(currPath, moduleName) {
   }
 
   var resolveError = null;
+  var paths = [];
   try {
-    return resolve.sync(moduleName, {
-      basedir: path.dirname(currPath),
-      extensions: ['.js', '.json']
-    });
+    if (NODE_PATH) {
+      if (NODE_PATH.indexOf(process.cwd()) !== -1) {
```

Comment:
This should probably be more specific by verifying that the `cwd()` is actually predicate of `NODE_PATH`. For example `NODE_PATH.substr(0, cwd.length) === cwd`